internal/engine/upper_test.go

```diff
@@ -3407,6 +3407,23 @@ func (f *testFixture) Start(manifests []model.Manifest, watchFiles bool, initOpt
 	f.startWithInitManifests(nil, manifests, watchFiles, initOptions...)
 }
 
+func (f *testFixture) StartAndWaitForInitialBuilds(manifests []model.Manifest, initOptions ...initOption) {
+	f.startWithInitManifests(nil, manifests, true, initOptions...)
+	if f.b.completeBuildsManually {
+		for i := range manifests {
+			f.b.completeBuild(i + 1) // builds are 1-indexed
+		}
+	}
+
+	expectedTargs := make([]model.ImageTarget, len(manifests))
```

Comment:
this check doesn't seem generally correct, because it assumes 1 image target per manifest

can you use the waitForCompletedBuildCount() check, for consistency with other tests?

Reply:
good call, will do